src/Modules/Authontication/Components/ChangePass/ChangePass.tsx

```diff
@@ -1,8 +1,170 @@
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { PasswordValidation } from "../../../../constans/VALIDATIONS";
+import axios from "axios";
+import { AuthorizedToken, USERS_URLs } from "../../../../constans/END_POINTS";
+import { toast } from "react-toastify";
 
 export default function ChangePass() {
+  const [showOldPassword, setShowOldPassword] = useState(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    getValues,
+  } = useForm({ mode: "onBlur" });
+
+  const toggleVisibility = (setterFunction: any) => {
+    setterFunction((prevState: any) => !prevState);
+  };
+
+  const onSubmit =  async(data: any) => {
+    try {
+      const response: any =await axios.put(USERS_URLs.ChangePassword, AuthorizedToken, data);
+      toast.success(
+        response.data.message || "Your password has been successfully changed!"
+      );
+    } catch (error: any) {
+      toast.error(
+        error.message || "Password change unsuccessful. Please try again"
+      );
+    }
+  };
+
   return (
-    <h1 className='text-white'>ChangePass</h1>
-  
-  )
+    <>
+      <div className="auth-title my-4">
+        <p className="text-white">Welcome to PMS</p>
+        <h3 className="main-color title">
+          <span className="frist-ch position-relative">C</span>hange Password
+        </h3>
+      </div>
+
+      <form className="container-form" onSubmit={handleSubmit(onSubmit)}>
+        <div className="my-4">
+          <label className="main-color my-1">Old Password</label>
+          <div className="input-group">
+            <input
+              type={showOldPassword ? "text" : "password"}
+              className="form-control form-input"
+              placeholder="Enter your Old Password"
+              aria-label="oldPassword"
+              {...register("oldPassword", PasswordValidation)}
+            />
+            <button
+              onMouseDown={(e) => e.preventDefault()}
+              onMouseUp={(e) => e.preventDefault()}
+              type="button"
+              onClick={() => toggleVisibility(setShowOldPassword)}
+              className="input-group-text bg-transparent border-0"
+            >
+              <span className="sr-only">
+                {showOldPassword ? "hide password" : "show password"}
+              </span>
+              <i
+                className={
+                  showOldPassword
+                    ? "fa-solid text-white fa-eye"
+                    : "fa-solid text-white fa-eye-slash"
+                }
+              ></i>
+            </button>
+          </div>
+          {errors.oldPassword && (
+            <span className="text-danger">
+              {String(errors.oldPassword.message)}
+            </span>
+          )}
+        </div>
+
+        <div className="my-4">
```

Comment:
We can abstract the password input into a component and then re-use it with its own internal logic like handling password visibility and so on.